parser-typechecker/tests/Unison/Test/CodebaseInit.hs

```diff
@@ -37,18 +41,35 @@ test = scope "Codebase.Init" $ tests
     [ scope "a v2 codebase should be opened" do
         tmp <- io (Temp.getCanonicalTemporaryDirectory >>= flip Temp.createTempDirectory "ucm-test")
         cbInit <- io initMockWithCodebase
-        res <- io (CI.openOrCreateCodebase cbInit "ucm-test" (Specified tmp))
+        res <- io (CI.openOrCreateCodebase cbInit "ucm-test" (Specified (DontCreateWhenMissing tmp)))
         case res of 
           CI.OpenedCodebase _ _ -> expect True
           _ -> expect False
     , scope "a v2 codebase should be *not* created when one does not exist at the Specified dir" do
         tmp <- io (Temp.getCanonicalTemporaryDirectory >>= flip Temp.createTempDirectory "ucm-test")
         cbInit <- io initMockWithoutCodebase
-        res <- io (CI.openOrCreateCodebase cbInit "ucm-test" (Specified tmp) )
+        res <- io (CI.openOrCreateCodebase cbInit "ucm-test" (Specified (DontCreateWhenMissing tmp)))
         case res of 
           CI.Error _ CI.NoCodebaseFoundAtSpecifiedDir -> expect True
           _ -> expect False
     ] 
+  , scope "*with* a --codebase-create flag" $ tests
+    [  scope "a v2 codebase should be created when one does not exist at the Specified dir" do
+        tmp <- io (Temp.getCanonicalTemporaryDirectory >>= flip Temp.createTempDirectory "ucm-test")
+        cbInit <- io initMockWithoutCodebase
+        res <- io (CI.openOrCreateCodebase cbInit "ucm-test" (Specified (CreateWhenMissing tmp)))
+        case res of 
+          CI.CreatedCodebase _ _ -> expect True
+          _ -> expect False
```

Comment:
For future reference you can use `ok` in place of `expect True`, and `fail "msg"` or `crash "msg"` instead of `expect False`.